src/PdoConnection.php

```diff
@@ -248,7 +268,17 @@ public function rollBack()
      */
     public function commit()
     {
-        return $this->connect()->commit();
+        if (--$this->transactionCount === 0) {
+            if (!$this->rollbackNeeded) {
+                return $this->connect()->commit();
+            }
+
+            $this->rollbackNeeded = false;
+            $this->connect()->rollback();
+            return false;
+        }
+
+        return ($this->transactionCount >= 0 && !$this->rollbackNeeded);
```

Comment:
The return is meant to signal whether an error occurred.   In the case were we have not reached the outer transaction, and none of the inner transactions have tried to rollback, we return true for success.